src/promptflow/promptflow/tracing/_utils.py

```diff
@@ -59,18 +58,25 @@ def serialize(value: object, remove_null: bool = False, serialization_funcs: Dic
 
 
 def get_input_names_for_prompt_template(template_str):
-    input_names = []
-    env = Environment()
-    template = env.parse(template_str)
-    input_names.extend(sorted(meta.find_undeclared_variables(template), key=lambda x: template_str.find(x)))
+    try:
+        # We need to parse jinja template only when the promptflow is installed and run flow with PromptTemplate
+        # type input, so using try-catch to avoid the dependency of jinja2 when it's not needed.
+        from jinja2 import Environment, meta
```

Comment:
not sure simply jinja2 or promptflow.core (I don't know the precise path...)?

if a user installs jinja2, but not from promptflow, is the below lines executed expected?

Reply:
1. It should be jinja2.
2. We need to parse jinja, only if the user installed the promptflow and used PromptTemplate type as input.